src/types/index.ts

```diff
@@ -1 +1,16 @@
 export type Command = () => void
+export interface WatchOrderPlacementParams {
+  // TODO: Try to use instead of "any" the autogenerated type from the ABI (data from the event)
+  onNewOrder: (data: any) => void
+  onError: (error: Error) => void
+}
+
+export interface DfusionRepo {
```

Comment:
The repo in the name refers to what kind of repository?

Asking because every time I read the name I think about a github repo :)

Reply:
Repo refers to a repository pattern, to abstract away the access to a datasource. In this case a blockchain one. It's kind of similar to the Dao pattern I'm sure you know since you worked a lot in Java land :) 
https://deviq.com/repository-pattern/

In the web, we created a layer call API, that in reality applies a similar concept. I don't care that much about the naming convention, so I'm open to suggestion. I care about defining just the abstraction a layer itself.

Reply:
No need to change the name, just clarifying.